### ISSUE TITLE
Error parsing undocumented API response

### ISSUE THREAD

**Issue by ibddot** [wez/govee2mqtt#6]
Starting the add-on gets error parsing undocumented API response `1: unknown field dspVersionSoft` and quits. 
I am running HAOS and followed the instructions to install this add-on.
Logs below. 

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
GOVEE_API_KEY=XXXX
GOVEE_MQTT_HOST=core-mosquitto
GOVEE_EMAIL=XXXX 
GOVEE_MQTT_USER=addons
GOVEE_MQTT_PORT=1883
++ cd /app
++ exec /app/govee serve
[2024-01-04T08:22:42 INFO  govee::commands::serve] Starting service. version 2024.01.03-63439388
[2024-01-04T08:22:42 INFO  govee::commands::serve] Querying platform API for device list
[2024-01-04T08:22:43 INFO  govee::commands::serve] Querying undocumented API for device + room list
Error: request status 200: OK

Caused by:
    0: parsing response as json: {"devices":[{"groupId":0,"device":"2A:64:F8:4F:C0:2E:56:AA","sku":"H706A","spec":"","versionHard":"3.04.01","versionSoft":"1.00.14","deviceName":"Permanent Lights Pro","deviceExt":{"deviceSettings":"{\"wifiName\":\"XXXX\",\"address\":\"F8:4F:C0:2E:56:AA\",\"bleName\":\"Govee_H706A_56AA\",\"topic\":\"GD/9b546caecc80eef27478fd5839453d27\",\"wifiMac\":\"60:74:F4:85:45:D4\",\"pactType\":1,\"pactCode\":1,\"wifiSoftVersion\":\"1.01.17\",\"wifiHardVersion\":\"1.04.01\",\"ic\":55,\"secretCode\":\"XXXX=\",\"boilWaterCompletedNotiOnOff\":1,\"completionNotiOnOff\":1,\"autoShutDownOnOff\":1,\"dspVersionSoft\":\"13\",\"supportBleBroadV3\":false,\"segmentCount\":11,\"sku\":\"H706A\",\"device\":\"2A:64:F8:4F:C0:2E:56:AA\",\"deviceName\":\"Permanent Lights Pro\",\"versionHard\":\"3.04.01\",\"versionSoft\":\"1.00.14\",\"playState\":false}","lastDeviceData":"{\"online\":false}","extResources":"{\"skuUrl\":\"https://d1f2504ijhdyjw.cloudfront.net/sku-img/110c816f749f75640260511c333a1065-add_list_type_device_706a.png\",\"headOnImg\":\"https://d1f2504ijhdyjw.cloudfront.net/sku-img/8e57eab8d1c8c6d9f280be7e4460b0d2-706a_light_title_on.png\",\"headOnImgNew\":\"\",\"headOffImg\":\"https://d1f2504ijhdyjw.cloudfront.net/sku-img/a4e9d925c3c788fc25adde52e44349a8-706a_light_title_off.png\",\"headOffImgNew\":\"\",\"ext\":\"\",\"ic\":120}"},"pactType":1,"pactCode":1,"share":0,"deviceId":17753321,"goodsType":180,"attributesId":49,"supportScene":0}],"groups":[{"groupId":16153650,"groupName":"Bedroom"},{"groupId":16153651,"groupName":"Living Room"}],"sort":[],"sortTime":0,"message":"","status":200}
    1: unknown field `dspVersionSoft`, expected one of `wifiName`, `address`, `bleName`, `topic`, `wifiMac`, `pactType`, `pactCode`, `wifiSoftVersion`, `wifiHardVersion`, `ic`, `ic_sub_1`, `ic_sub_2`, `secretCode`, `boilWaterCompletedNotiOnOff`, `completionNotiOnOff`, `autoShutDownOnOff`, `sku`, `device`, `deviceName`, `versionHard`, `versionSoft`, `playState` at line 1 column 371

Stack backtrace:
   0: anyhow::context::<impl anyhow::Context<T,E> for core::result::Result<T,E>>::with_context
   1: govee::undoc_api::GoveeUndocumentedApi::get_device_list::{{closure}}
   2: govee::commands::serve::ServeCommand::run::{{closure}}
   3: govee::Args::run::{{closure}}
   4: tokio::runtime::park::CachedParkThread::block_on
   5: tokio::runtime::context::runtime::enter_runtime
   6: govee::main
   7: std::sys_common::backtrace::__rust_begin_short_backtrace
   8: std::rt::lang_start::{{closure}}
   9: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/core/src/ops/function.rs:284:13
  10: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  11: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  12: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  13: std::rt::lang_start_internal::{{closure}}
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/rt.rs:148:48
  14: std::panicking::try::do_call
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:552:40
  15: std::panicking::try
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panicking.rs:516:19
  16: std::panic::catch_unwind
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/panic.rs:142:14
  17: std::rt::lang_start_internal
             at /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/rt.rs:148:20
  18: std::rt::lang_start
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by wez** [comment]
Thanks for sharing this. I think this should be generally resolved by ea42fcc284448bbb73ab0488c36cfb619f86d901
If you update the addon it should hopefully work!

**Comment by ibddot** [comment]
Thanks. Now it is working.
